src/main/java/picard/sam/ValidateSamFile.java

```diff
@@ -133,75 +133,117 @@ public static void main(final String[] args) {
         System.exit(new ValidateSamFile().instanceMain(args));
     }
 
+    /**
+     * Return tyeps for doWork()
+     */
+    enum ReturnTypes {
+        FAILED(-1),     // failed to complete execution
+        SUCCESSFUL(0),  // ran successfully
+        WARNINGS(1),    // warnings but no errors
```

Comment:
Warnings does not mean that the file is not a valid SAM file, but that it could have some problems (such as no read groups).

Although there is an option to ignore warnings, I can imagine a situation where you want to keep track of the possible problems with the file but continue the pipeline as the file does not contain "ERRORS". Although it may be not the recommended way of doing this, I guess that it is a fair use case...